.eslintrc.js

```diff
@@ -39,85 +19,223 @@ module.exports = {
+        "react-hooks/rules-of-hooks": ERROR, // Check rules of Hooks
+        "react-hooks/exhaustive-deps": WARN, // Warn useMemo, useEffect dependencies
+
+        // Rules for React.
+        "react/destructuring-assignment": OFF, // Allow use of dot notation, for example user.id (airbnb rule)
+        "react/display-name": OFF,
+        "react/function-component-definition": [
+          errorIfStrict,
+          {
+            namedComponents: "arrow-function", // Allow named components with arrow functions (airbnb rule)
+          },
+        ],
+        "react/jsx-filename-extension": [1, { extensions: [".tsx"] }], // Allow JSX in TSX file (airbnb rule)
+        "react/jsx-props-no-spreading": [
+          errorIfStrict,
+          {
+            custom: "ignore",
+            explicitSpread: "ignore",
+          },
+        ],
+        "react/prop-types": OFF, // (airbnb rule)
+        "react/react-in-jsx-scope": OFF, // Disable because there is no need to import React in React 17+ (airbnb rule)
+        "react/require-default-props": OFF, // Allow not setting defaults for props (airbnb rule)
+        "react/sort-comp": [
+          errorIfStrict,
+          { order: ["everything-else", "render"] },
+        ],
+        "react/no-unstable-nested-components": OFF, // This rule should be removed as part of EVG-17265.
+      },
+      parserOptions: {
+        project: ["./tsconfig.json"],
+      },
+    },
+    // For Jest files.
+    {
+      files: ["src/**/*.test.ts", "src/**/*.test.tsx"],
+      extends: ["plugin:testing-library/react", "plugin:jest/all"],
+      rules: {
+        "@typescript-eslint/unbound-method": OFF,
+        "jest/max-expects": [
+          WARN,
+          {
+            max: 10,
+          },
+        ],
+        "jest/no-hooks": OFF,
+        "jest/no-mocks-import": OFF,
+        "jest/prefer-expect-assertions": OFF,
+        "jest/unbound-method": OFF,
+      },
+    },
+    // For Storybook files.
+    {
+      files: ["src/**/*.stories.@(ts|tsx|js|jsx|mjs|cjs)"],
+      extends: ["plugin:storybook/recommended"],
+      rules: {
+        "storybook/story-exports": ERROR,
+        "storybook/no-stories-of": ERROR,
+        "storybook/no-redundant-story-name": WARN,
+        "storybook/csf-component": WARN, // Enables support for auto generated prop controls and documentation
+      },
+    },
+    // For Cypress files.
+    {
+      files: ["cypress/**/*.ts"],
+      extends: ["plugin:cypress/recommended"],
+      parserOptions: {
+        project: "cypress/tsconfig.json",
+      },
+    },
+    // For GraphQL files.
     {
-      files: ["*.graphql"],
-      parser: "@graphql-eslint/eslint-plugin",
-      plugins: ["@graphql-eslint"],
+      files: ["src/**/*.graphql"],
+      extends: "plugin:@graphql-eslint/operations-recommended",
       rules: {
-        "prettier/prettier": "error",
+        "@graphql-eslint/naming-convention": OFF, // This should be turned on eventually.
```

Comment:
This should actually be turned on, but it will cause like ~200 file changes so I left it out.

Reply:
Should we make a ticket for this and reference it here?

Reply:
Created [EVG-19316](https://jira.mongodb.org/browse/EVG-19316)!